timemachine/fe/single_topology.py

```diff
@@ -1109,66 +1085,80 @@ def setup_intermediate_state(
             dst_system.bond,
             lamb,
             interpolate.align_harmonic_bond_idxs_and_params,
-            interpolate_harmonic_bond_params_fxn or self.default_interpolate_harmonic_bond_params_fxn,
+            partial(
+                interpolate_harmonic_bond_params,
+                k_min=0.1,  # ~ BOLTZ * (300 K) / (5 nm)^2
+                lambda_min=0.0,
+                lambda_max=lambda_angles,
+            ),
         )
 
         angle = self._setup_intermediate_bonded_term(
             src_system.angle,
             dst_system.angle,
             lamb,
             interpolate.align_harmonic_angle_idxs_and_params,
-            interpolate_harmonic_angle_params_fxn or self.default_interpolate_harmonic_angle_params_fxn,
+            partial(
+                interpolate_harmonic_angle_params,
+                k_min=0.05,  # ~ BOLTZ * (300 K) / (2 * pi)^2
+                lambda_min=lambda_angles,
+                lambda_max=lambda_torsions,
+            ),
         )
 
         torsion = self._setup_intermediate_bonded_term(
             src_system.torsion,
             dst_system.torsion,
             lamb,
             interpolate.align_torsion_idxs_and_params,
-            interpolate_periodic_torsion_params_fxn or self.default_interpolate_periodic_torsion_params_fxn,
+            partial(
+                interpolate_periodic_torsion_params,
+                lambda_min=lambda_torsions,
+                lambda_max=1.0,
+            ),
         )
 
         nonbonded = self._setup_intermediate_nonbonded_term(
             src_system.nonbonded,
             dst_system.nonbonded,
             lamb,
             interpolate.align_nonbonded_idxs_and_params,
-            interpolate_nonbonded_params_fxn or self.default_interpolate_nonbonded_params_fxn,
+            interpolate.linear_interpolation,
         )
 
         chiral_atom = self._setup_intermediate_bonded_term(
             src_system.chiral_atom,
             dst_system.chiral_atom,
             lamb,
             interpolate.align_chiral_atom_idxs_and_params,
-            interpolate_chiral_atom_params_fxn or self.default_interpolate_chiral_atom_params_fxn,
+            interpolate.linear_interpolation,
         )
 
         chiral_bond = self._setup_intermediate_chiral_bond_term(
             src_system.chiral_bond,
             dst_system.chiral_bond,
             lamb,
-            interpolate_chiral_bond_params_fxn or self.default_interpolate_chiral_bond_params_fxn,
+            interpolate.linear_interpolation,
         )
 
         return system.VacuumSystem(bond, angle, torsion, nonbonded, chiral_atom, chiral_bond)
 
     def _parameterize_host_guest_nonbonded(self, lamb, host_nonbonded):
         # Parameterize nonbonded potential for the host guest interaction
 
-        guest_exclusions = []
-        guest_scale_factors = []
+        guest_exclusions_ = []
```

Comment:
did you intend to add the extra `_` at the end?

Reply:
(is this to avoid aliasing/re-assignment with the guest_exclusions defined below?)

Reply:
Correct, the motivation was to fix mypy errors of the form

```
error: Incompatible types in assignment (expression has type "ndarray[Any, dtype[floating[_64Bit]]]", variable has type "List[Tuple[float, float]]")
```

(this particular case could actually be reverted; I had changed this when the signature of `_parameterize_host_guest_nonbonded` had an annotated type, which triggered mypy to check the function body

Reply:
Reverted this instance in fd37b16 to minimize the diff since ended up not being strictly necessary

Reply:
I'm okay either way! Up to you